redbeams/src/main/java/com/sequenceiq/redbeams/controller/v4/databaseserver/DatabaseServerV4Controller.java

```diff
@@ -251,4 +252,16 @@ public UpgradeDatabaseServerV4Response upgrade(@TenantAwareParam @NotEmpty @Vali
         UpgradeDatabaseRequest upgradeDatabaseRequest = upgradeDatabaseServerV4RequestConverter.convert(request);
         return upgradeDatabaseServerV4ResponseConverter.convert(redbeamsUpgradeService.upgradeDatabaseServer(databaseServerCrn, upgradeDatabaseRequest));
     }
+
+    @Override
+    @InternalOnly
+    public void validateUpgrade(@TenantAwareParam @NotEmpty @ValidCrn(resource = CrnResourceDescriptor.DATABASE_SERVER) String crn,
+            @Valid @NotNull UpgradeDatabaseServerV4Request request) {
+        UpgradeDatabaseRequest upgradeDatabaseRequest = upgradeDatabaseServerV4RequestConverter.convert(request);
+        try {
+            redbeamsUpgradeService.validateUpgradeDatabaseServer(crn, upgradeDatabaseRequest);
+        } catch (Exception exception) {
+            throw new BadRequestException(exception.getMessage(), exception);
+        }
```

Comment:
this mapping is a bit strange to me. I would expect that this would return with a validation result, and it would contain if the validation is successful or not. and this would throw badrequest in case the request is wrong, like no db crn is present.
I'm not sure about how we should handle provider side issues. Maybe those should just cause validation failures, or we should just let the upgrade to continue